aten/src/ATen/native/Loss.cpp

```diff
@@ -339,9 +339,6 @@ Tensor smooth_l1_loss(const Tensor& input, const Tensor& target, const int64_t r
 
 Tensor& smooth_l1_loss_out(const Tensor& input, const Tensor& target, int64_t reduction, double beta, Tensor& result) {
   TORCH_CHECK(beta >= 0, "smooth_l1_loss does not support negative values for beta.")
-  if (beta == 0) {
```

Comment:
why do we wanna remove this?

Reply:
We might be able to put it back after https://github.com/pytorch/pytorch/pull/65681#issuecomment-928836372 is resolved. For the other two cases, we can't keep them, because they call into functions that will no longer be available (`l1_loss_backward_out` and `l1_loss_backward`). 

Note that this is only a shortcut and does not change the output in case it is left out. It might be a perf regression though. I'm going to benchmark.

Reply:
I've benchmarked with [this script]() and these are the results:

```
[---------- smooth_l1_loss with beta=0 ---------]
                             |  master  |    PR  
1 threads: --------------------------------------
      cpu / ()               |      30  |      24
      cpu / (10000000,)      |   56700  |   59000
      cpu / (10000, 10000)   |  626000  |  610000
      cuda / ()              |      30  |      25
      cuda / (10000000,)     |   60000  |   58000
      cuda / (10000, 10000)  |  612000  |  590000
2 threads: --------------------------------------
      cpu / ()               |      29  |      24
      cpu / (10000000,)      |   43000  |   49000
      cpu / (10000, 10000)   |  530000  |  490000
4 threads: --------------------------------------
      cpu / ()               |      29  |      24
      cpu / (10000000,)      |   39000  |   42000
      cpu / (10000, 10000)   |  500000  |  450000

Times are in microseconds (us).
```

- CUDA execution is faster than before
- CPU execution is slower for large 1D inputs, but faster for multi-dimensional inputs.

In all cases we are talking about a few milliseconds for quite large inputs.